students/charlie_england/session07/html_render.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+# This is the framework for the base class
+class Element(object):
+    tag = "html"
+
+    def __init__(self, content=None,**attributes):
+        self.content_lst = [f"<{self.tag}>",]
+        self.__dict__.update(attributes)
```

Comment:
careful here! this is kinda black magic! Yes, it can work, but you have updated the namespace of the entire instance -- now your attributes are tangled up with the regular instance attributes: `self.content_lst` for example. And there may be more in the future.

Better to keep the tag attributes in their own separate dict.